com.syntevo.SmartGit.json

```diff
@@ -0,0 +1,85 @@
+{
+  "app-id": "com.syntevo.SmartGit",
+  "command": "smartgit",
+  "runtime": "org.gnome.Platform",
```

Comment:
Does it it actually need that or is org.freedesktop.Platform enough?

Reply:
SmartGit (like any other SWT based Java application) needs a 64-bit desktop environment supporting GTK3 (currently GTK2 might also work but has been said to be dropped quickly).

We are not experienced with Flatpak - will it install to a fixed or to an arbitrary location? If the latter, how to get Git bundled, too (we haven't succeeded in building it statically linked in a portable manner).

Reply:
@eszlari, the application also "works" with `Freedesktop`, but some UI components are rendered incorrectly, that's why I chose `GNOME` in the end.

@syntevo, Flatpaks are executed in a sandboxed environment. Depending on the access provided, they would be able to "see" most of the host's filesystem. I gave access to `host` to this one, because some people work on different partitions, and also to be able to access `Git` on the host's operating system. Git can also be bundled, but since the package can access it on the host's filesystem, I didn't do it – I do use mine in such a way, that's how I tested it.

Reply:
@eszlari, I tested with `Freedesktop` versions `1.6` and `18.08`. Same result. Anyway, I'll give it another try now that I have this process automated in a VM.

On bundling Git, I still think that it's an external dependency, but it's straightforward, so I'll bundle it. Are there any rules about when to upgrade which? Or those dependencies are upgrades/updated every time a new release of `SmartGit` is made to their latest stable one?

Reply:
When you update is up to you as a maintainer, but if there is a security bug, you should obviously update as fast as possible.

Reply:
This change is done; I can use the bundled `git` located in `/app/bin/git` now.

I gave it another whirl with both versions of `Freedesktop`: `1.6` and `18.08`, but the combo-boxes and sub-menus are not getting rendered properly.

Reply:
Maybe it's because they bundle Java 8. You could try 10, see e.g.:
https://github.com/flathub/org.jdownloader.JDownloader/blob/master/org.jdownloader.JDownloader.yaml

Reply:
...looks perfect now. I'll change to `Freedesktop 1.6` since I don't see an `18.08` version for `org.freedesktop.Sdk.Extension.openjdk10`; once @syntevo folks upgrades the JRE that's shipped with `SmartGit` to version `10` (or the upcoming `11`) I'll delete the extension and fallback to the provided one.